src/thor/timedep_forward.cc

```diff
@@ -0,0 +1,285 @@
+#include <iostream> // TODO remove if not needed
+#include <map>
+#include <algorithm>
+#include "baldr/datetime.h"
+#include "midgard/logging.h"
+#include "thor/timedep.h"
+
+using namespace valhalla::baldr;
+using namespace valhalla::sif;
+
+namespace valhalla {
+namespace thor {
+
+constexpr uint64_t kInitialEdgeLabelCount = 500000;
+
+// Default constructor
+TimeDepForward::TimeDepForward()
+    : AStarPathAlgorithm() {
+  mode_ = TravelMode::kDrive;
+  travel_type_ = 0;
+  adjacencylist_ = nullptr;
+  max_label_count_ = std::numeric_limits<uint32_t>::max();
+}
+
+// Destructor
+TimeDepForward::~TimeDepForward() {
+  Clear();
+}
+
+// Expand from the node along the forward search path. Immediately expands
+// from the end node of any transition edge (so no transition edges are added
+// to the adjacency list or EdgeLabel list). Does not expand transition
+// edges if from_transition is false.
+void TimeDepForward::ExpandForward(GraphReader& graphreader,
```

Comment:
This function is mostly a duplicate of `ExpandForward` from `astar.cc` - could this be templatized and that code re-used?

Reply:
this will likely be changing a bit more as this is the first template. If after hooking in all the time logic and updates to costing seems like it is still close then can see about re-use.